packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -196,29 +199,153 @@ class CupertinoAlertDialog extends StatelessWidget {
       children.add(const Padding(padding: const EdgeInsets.only(top: 8.0)));
     }
 
+    return new ClipRRect(
+      borderRadius: const BorderRadius.only(
+        topLeft: const Radius.circular(12.0),
+        topRight: const Radius.circular(12.0),
+      ),
+      child: new Container(
+        color: _kDialogForegroundColor,
+        child: new Column(
+          mainAxisSize: MainAxisSize.min,
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: children,
+        ),
+      ),
+    );
+  }
+
+  Widget _buildActions() {
+    Widget actionSection = new Container();
     if (actions.isNotEmpty) {
-      final Widget actionSection = new _CupertinoAlertActionSection(
+      actionSection = new _CupertinoAlertActionSection(
         children: actions,
         scrollController: actionScrollController,
       );
-      children.add(
-        new Flexible(child: actionSection),
-      );
     }
 
-    return new Padding(
-      padding: const EdgeInsets.symmetric(vertical: _kEdgePadding),
-      child: new CupertinoDialog(
-        child: new Column(
-          mainAxisSize: MainAxisSize.min,
-          crossAxisAlignment: CrossAxisAlignment.stretch,
-          children: children,
+    return new ClipRRect(
+      borderRadius: const BorderRadius.only(
+        bottomLeft: const Radius.circular(12.0),
+        bottomRight: const Radius.circular(12.0),
+      ),
+      child: actionSection,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Container(
+        width: _kCupertinoDialogWidth,
+        padding: const EdgeInsets.symmetric(vertical: _kEdgePadding),
+        child: new CustomMultiChildLayout(
+          delegate: new _CupertinoAlertDialogLayoutDelegate(
+            actionsCount: actions.length,
+          ),
+          children: <Widget>[
+            new LayoutId(
+              id: _AlertDialogSection.blur,
+              child: _buildBlurBackground(),
```

Comment:
Maybe you can just wrap the container at 239 with the contents of this function instead of having it have its own layout id.

Reply:
The container is actually the maximum possible height for a dialog.  Remember that issue where the dialog was always the height of the screen because the parent size can't depend on the children?  The way I'm getting around that is by manually laying out all the individual pieces of the dialog, including the blur area.